app/models/katello/concerns/smart_proxy_extensions.rb

```diff
@@ -40,7 +40,7 @@ module SmartProxyExtensions
                  :source     => :lifecycle_environment
 
         has_many :content_facets, :class_name => "::Katello::Host::ContentFacet", :foreign_key => :content_source_id,
-                                  :inverse_of => :content_source
+                                  :inverse_of => :content_source, :dependent => :nullify
```

Comment:
i made  a note here to look into this more:  http://projects.theforeman.org/issues/21021